stl/src/vector_algorithms.cpp

```diff
@@ -2916,10 +2916,316 @@ namespace {
         return _Result;
     }
 
-    namespace __std_find_first_of {
+#ifndef _M_ARM64EC
+    namespace __std_find_meow_of {
+        enum class _Strategy { _No_bitmap, _Scalar_bitmap, _Vector_bitmap };
+
+        bool _Product_fits_threshold(const size_t _Px1, const size_t _Px2, const size_t _Tx) noexcept {
+#ifdef _WIN64
+            long long _Rx;
+            return _mul_overflow_i64(_Px1, _Px2, &_Rx) && static_cast<size_t>(_Rx) < _Tx;
+#else // ^^^ defined(_WIN64) / !defined(_WIN64)
+            unsigned int _Rx0;
+            unsigned int _Rx1;
+            return _mul_full_overflow_u32(_Px1, _Px2, &_Rx0, &_Rx1) && _Rx0 < _Tx;
+#endif
+        }
+
+        template <class _Ty>
+        _Strategy _Pick_strategy_avx(const size_t _Count1, const size_t _Count2) noexcept {
+            if constexpr (sizeof(_Ty) == 1) {
+                if (_Count1 <= 15 || _Product_fits_threshold((_Count1 + 15) / 16, (_Count2 + 15) / 16, 60)) {
+                    return _Strategy::_No_bitmap;
+                } else if (_Count1 * 1ull > _Count2 * 5ull) {

```

Comment:
The `ull` suffix is extending potentially large 32-bit counter to 64-bit in 32-bit mode.
In 64-bit mode there's nowhere to extend, but there's no way to address that many elements anyway.